### PR TITLE
perf: optimize shplonk verification

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR optimizes `SHPlonk` verification by creating `G2Prepared` in advance and calling `Pairing()` just once.

### CODE REVIEW

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: tachyon/crypto/commitments/kzg/shplonk.h [hunk 1]
```diff
@@ -40,6 +41,7 @@ class SHPlonk final : public UnivariatePolynomialCommitmentScheme<
       SHPlonk<Curve, MaxDegree, Commitment>>;
   using G1Point = typename Curve::G1Curve::AffinePoint;
   using G2Point = typename Curve::G2Curve::AffinePoint;
+  using G2Prepared = typename Curve::G2Prepared;
```
Comment (Insun35) [inline tachyon/crypto/commitments/kzg/shplonk.h:44]: I thought you modified the pairing logic to be able to use only w/ `G1Point` and `G2 Point` w/o `G2Prepared` in this [commit](https://github.com/kroma-network/tachyon/pull/222/commits/779eb27b82cc4f78d26ff2289d5653606670300d#diff-d51496b909572d7b364c5dedca54853dba6ac3af77c865bc8ce5c511ebb569b4).
Is there any reason to use it again?

Comment (chokobole) [reply tachyon/crypto/commitments/kzg/shplonk.h:44]: `G2Prepared` looks like an expensive operation and in our case, the values are fixed so we computed these values in advance and we uses them.

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Pull request merged**